zap/src/output/luau/mod.rs

```diff
@@ -18,69 +15,526 @@ pub trait Output {
 		self.push("\n");
 	}
 
-	fn push_stmt(&mut self, stmt: &Stmt) {
-		if matches!(stmt, Stmt::ElseIf(..) | Stmt::Else | Stmt::End) {
-			self.dedent();
+	fn push_line_indent(&mut self, s: &str) {
+		self.push_line(s);
+		self.indent();
+	}
+
+	fn push_dedent_line(&mut self, s: &str) {
+		self.dedent();
+		self.push_line(s);
+	}
+
+	fn push_dedent_line_indent(&mut self, s: &str) {
+		self.dedent();
+		self.push_line(s);
+		self.indent();
+	}
+
+	fn push_range_check(&mut self, range: Range, val: &str) {
+		if let Some(exact) = range.exact() {
+			self.push_line(&format!("assert({val} == {exact})"));
```

Comment:
Needs second arg for `assert` - see https://discord.com/channels/385151591524597761/906369439262461992/1195943890340040764